arroyo/processing/processor.py

```diff
@@ -243,7 +222,6 @@ def on_partitions_assigned(partitions: Mapping[Partition, int]) -> None:
                         "Partition assignment while processing strategy active"
                     )
                     _close_strategy()
-                _create_strategy(partitions)
```

Comment:
Should we assert that `self.__processing_strategy` is None here? What would happen if we get an assign without a prior revocation? I know that doesn't actually happen today but the code right now is designed to work for incremental assignment as well, which might not be the case after this change.

Reply:
not sure i follow, we do close it on the line before exactly when the strategy is not None

Reply:
fwiw this part of rebalancing was actually broken with incremental assign, as we would create the strategy with the subset of partitions from this callback instead of the full assignment. with this pr we're now actually only calling `_create_strategy` with `consumer.tell()` which should always contain the full assignment